lib/spike.rb

```diff
@@ -0,0 +1,361 @@
+require 'forwardable'
+require 'json'
+require 'pathname'
+require 'active_support/inflector'
+
+module Spike
+  class Resource
```

Comment:
This is a small class that tries to figure out what resource we're dealing with based on the path.